lazyflow/utility/reorderAxesDecorator.py

```diff
@@ -0,0 +1,447 @@
+                self.childOp.InputB.connect(self.Input)
+                self.Output.connect(self.childOp.Output)
+    """
+    if '_reorder_exceptions' not in dir(cls):
+        cls._reorder_exceptions = []
+
+    cls = guard_methods(cls)
+
+    inputSlots = [s.name for s in cls.inputSlots
+                  if s.name not in cls._reorder_exceptions]
+    outputSlots = [s.name for s in cls.outputSlots
+                   if s.name not in cls._reorder_exceptions]
+
+    inner_prefix = 'inner_'
+    cls._inner_call = False  # use this variable to distinguish in __getattribute__, which slot to return
+
+    # change __init__ in order to squeeze in the opReorderAxes ops
+    old_init = getattr(cls, '__init__')
+    def guard(fn):
+        @functools.wraps(old_init)
+        def wrap(self, *args, graph=None, parent=None, **kwargs):
+            ret = fn(self, *args, graph=graph, parent=parent, **kwargs)
+            graph, parent = self.graph, self.parent
+            self._inner_call = False
+
+            self._opReorderInput = {}
+            for name in inputSlots:
+                self._opReorderInput[name] = OpReorderAxes(parent=self)
+            
+            self._opReorderOutput = {}
+            for name in outputSlots:
+                self._opReorderOutput[name] = OpReorderAxes(parent=self)
+
+            return ret
+        return wrap
+
+    setattr(cls, '__init__', guard(old_init))
+
+    # change setupOutputs in order to connect the squeezed in opReorderAxes ops
+    old_setupOutputs = getattr(cls, 'setupOutputs')
+    def guard(fn):
+        @functools.wraps(old_setupOutputs)
+        def wrap(self, *args, **kwargs):
+            self._inner_call = False
+            for name in inputSlots:
+                self._opReorderInput[name].AxisOrder.setValue(self._internal_axes_order)
+
+                slot = self.__getattribute__(name)
+                self._opReorderInput[name].Input.connect(slot)
+
+            for name in outputSlots:
+                self._opReorderOutput[name].AxisOrder.setValue(cls._output_axes_order)
+                
+                slot = self.__getattribute__(name)
+                slot.connect(self._opReorderOutput[name].Output)
+
+            self._inner_call = True
+            ret = fn(self, *args, **kwargs)
+            self._inner_call = False
+
+            return ret
+        return wrap
+
+    setattr(cls, 'setupOutputs', guard(old_setupOutputs))
+
+    # change __getattribute__ in order to squeeze in the opReorderAxes ops
+    def __getattribute__(self, name):
+        if name in inputSlots:
+            if self._inner_call:
+                return self._opReorderInput[name].Output
+
+        elif name in outputSlots:
+            if self._inner_call:
+                return self._opReorderOutput[name].Input
+
+        return super(cls, self).__getattribute__(name)
+
+    setattr(cls, '__getattribute__', __getattribute__)
+    return cls
+
+
+if __name__ == '__main__':
```

Comment:
Would it be cleaner to extract this logic in its own test case?

Reply:
I guess so. My reasoning was that this decorator is only meant as a temporary tool. If we keep it around for a while, we should move this test logic.